site/src/pages/UserSettingsPage/NotificationsPage/NotificationsPage.tsx

```diff
@@ -0,0 +1,227 @@
+import Card from "@mui/material/Card";
+import Divider from "@mui/material/Divider";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import ListItemText, { listItemTextClasses } from "@mui/material/ListItemText";
+import Switch from "@mui/material/Switch";
+import Tooltip from "@mui/material/Tooltip";
+import { Fragment, type FC } from "react";
+import { useMutation, useQueries, useQueryClient } from "react-query";
+import {
+  notificationDispatchMethods,
+  selectTemplatesByGroup,
+  systemNotificationTemplates,
+  updateUserNotificationPreferences,
+  userNotificationPreferences,
+} from "api/queries/notifications";
+import type {
+  NotificationPreference,
+  NotificationTemplate,
+} from "api/typesGenerated";
+import { displaySuccess } from "components/GlobalSnackbar/utils";
+import { Loader } from "components/Loader/Loader";
+import { Stack } from "components/Stack/Stack";
+import { useAuthenticated } from "contexts/auth/RequireAuth";
+import { methodIcons, methodLabel } from "modules/notifications/utils";
+import { Section } from "../Section";
+
+type PreferenceSwitchProps = {
+  id: string;
+  disabled: boolean;
+  onToggle: (checked: boolean) => Record<string, boolean>;
+};
+
+const PreferenceSwitch: FC<PreferenceSwitchProps> = ({
+  id,
+  disabled,
+  onToggle,
+}) => {
+  const { user } = useAuthenticated();
+  const queryClient = useQueryClient();
+  const updatePreferences = useMutation(
+    updateUserNotificationPreferences(user.id, queryClient),
+  );
+
+  return (
+    <Switch
+      id={id}
+      size="small"
+      checked={!disabled}
+      onChange={async (_, checked) => {
+        await updatePreferences.mutateAsync({
+          template_disabled_map: onToggle(checked),
+        });
+        displaySuccess("Notification preferences updated");
+      }}
+    />
+  );
+};
+
+export const NotificationsPage: FC = () => {
+  const { user, permissions } = useAuthenticated();
+  const [disabledPreferences, templatesByGroup, dispatchMethods] = useQueries({
+    queries: [
+      {
+        ...userNotificationPreferences(user.id),
+        select: selectDisabledPreferences,
+      },
+      {
+        ...systemNotificationTemplates(),
+        select: (data: NotificationTemplate[]) => {
+          const groups = selectTemplatesByGroup(data);
+          return permissions.viewDeploymentValues
+            ? groups
+            : {
+                // Members only have access to the "Workspace Notifications" group
+                ["Workspace Events"]: groups["Workspace Events"],
+              };
+        },
+      },
+      notificationDispatchMethods(),
+    ],
+  });
+  const ready =
+    disabledPreferences.data && templatesByGroup.data && dispatchMethods.data;
+
+  return (
+    <Section
+      title="Notifications"
+      description="Configure notifications. Some may be disabled by the deployment administrator."
```

Comment:
Can we replace this copy with TODO so we don't forget to modify it before merging?
```suggestion
      description="Configure notifications. TODO."
```

Reply:
I don't like to have TODOs 😆 @stirby I would appreciate your copy skills.

Reply:
My suggestion: 

"Configure your notification preferences. Icons on the right of each notification indicate delivery method, either SMTP or Webhook."